x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserMlNodeService.java

```diff
@@ -99,6 +95,18 @@ public ElserMlNodeModel parseRequestConfig(
         return new ElserMlNodeModel(modelId, taskType, NAME, serviceSettingsBuilder.build(), taskSettings);
     }
 
+    private static String selectDefaultModelVersionBasedOnClusterArchitecture(Set<String> modelArchitectures) {
```

Comment:
this refactor wasn't necessary for this PR, but I had done it to try to display the required model variant, discovered I wasn't able to do that, and left in this refactor because it seems good anyway.